docs/creating-a-static-check.md

```diff
@@ -0,0 +1,203 @@
+   *
+   * @return array The categories for the check.
+   */
+  public function get_categories() {
+    return array( Check_Categories::CATEGORY_GENERAL );
+  }
+
+  /**
+   * Returns an associative array of arguments to pass to PHPCS.
+   *
+   * @return array An associative array of PHPCS CLI arguments.
+   */
+  protected function get_args() {
+    return array(
+      'extensions' => 'php',
+      'standard'   => 'WordPress',
+      'sniffs'     => 'WordPress.WP.CapitalPDangit',
+    );
+  }
+}
+```
+
+## Creating a new check using file search for strings / regular expressions
+
+A new class should be created for the static check, extending the `Abstract_File_Check` class. Here is an example:
+
+```php
+use WordPress\Plugin_Check\Checker\Check_Categories;
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Traits\Stable_Check;
+
+/**
+ * Check for detecting incorrect casing of the term "WordPress" (specifically "Wordpress") using string search in files.
+ *
+ * @since n.e.x.t
+ */
+class My_Custom_Check extends Abstract_File_Check {
+
+  use Stable_Check;
+
+  /**
+   * Gets the categories for the check.
+   *
+   * Every check must have at least one category.
+   *
+   * @since n.e.x.t
+   *
+   * @return array The categories for the check.
+   */
+  public function get_categories() {
+    return array( Check_Categories::CATEGORY_PLUGIN_REPO );
+  }
+
+  /**
+   * Check the "Wordpress" in files.
+   *
+   * @since n.e.x.t
+   *
+   * @param Check_Result $result The Check Result to amend.
+   * @param array        $files  Array of plugin files.
+   */
+  protected function check_files( Check_Result $result, array $files ) {
+    // Get all php files in the plugin.
+    $php_files = self::filter_files_by_extension( $files, 'php' );
+
+    // Check files for instances of the "Wordpress".
+    $file = self::file_str_contains( $php_files, 'Wordpress' );
+
+    if ( $file ) {
+      $result->add_message(
+        true,
+        __( 'Please spell "WordPress" correctly.', 'plugin-check' ),
+        array(
+          'code' => 'misspelled_wordpress_in_files',
+          'file' => $file,
+        )
+      );
+    }
+  }
+}
+```
+
+## Amending the check result object
+
+The check's `run()` method will hold all the logic to test the plugin and raise any warnings or errors that are found.
+
+The run method accepts an instance of the `Check_Result` class which is used to add errors and warnings to the results list.
+
+The warnings and errors are added via the `add_message()` method which accepts 3 parameters.
+
+- `$error (bool)` - Whether the message is an error or warning. True for error, false for warning.
+- `$message (string)` - The error/warning message.
+- `$args (array)` - Additional message arguements to add context.
+  - `$code (string)` - Violation code according to the message. Default empty string.
+  - `$file (string)` - The file in which the message occurred. Default empty string (unknown file).
```

Comment:
```suggestion
  - `$file (string)` - The file for which the message occurred. Default empty string (unknown file).
```

I am not sure if this is correct, but I assume it's the file that triggered the error message, not that the message was _in_ correct?